cmd/main.go

```diff
@@ -208,6 +209,17 @@ func main() {
 		os.Exit(1)
 	}
 
+	if availCrds[controller.MaintenanceModeCRDName] {
```

Comment:
I'm not sure why we need to involve available CRSs here. We don't have conditional watches at this point just a single decision on should we run the controller or not

Reply:
If we don't have the CRD for maintenanceMode created, the operator pod won't start as we have a new controller For MaintenanceMode.
If the CRD is created at a later point in time, we need to restart the process for ocs-client-op so that we start the maintenanceMode controller

Reply:
Yes, right we have a availCrds loaded at this position and we are reusing it

Reply:
Ok, in that case, you might have a different issue. what happens if the crd is not avialable at operator startup but becomes available later on? your controller is not running.

Do we have a mechanism in place to restart the operator in such cases to allow the new controller to come up?

Reply:
Yes, we implemented it with https://github.com/red-hat-storage/ocs-client-operator/pull/253

Reply:
ok, resolving